### ISSUE TITLE
Compilation error building latest master on MacOS

### ISSUE THREAD

**Issue by hodlonaut** [input-output-hk/jormungandr#1376]
On branch master
Your branch is up to date with 'origin/master'.

git pull
Already up to date.

git submodule update --recursive

Compiling r2d2 v0.8.7
   Compiling http-body v0.1.0
   Compiling tower-request-modifier v0.1.0
   Compiling parking_lot v0.7.1
   Compiling multihash v0.8.0
   Compiling tokio-reactor v0.1.11
   Compiling serde_derive v1.0.104
   Compiling thiserror-impl v1.0.9
   Compiling v_escape_derive v0.5.6
   Compiling crossbeam-deque v0.7.2
   Compiling cid v0.3.1
   Compiling tempfile v3.1.0
   Compiling ed25519-dalek v1.0.0-pre.3
   Compiling tokio-tcp v0.1.3
   Compiling tokio-udp v0.1.5
   Compiling tokio-uds v0.2.5
   Compiling tokio-signal v0.2.7
   Compiling tokio-threadpool v0.1.17
   Compiling crossbeam v0.7.3
   Compiling multiaddr v0.3.1
   Compiling failure v0.1.6
   Compiling http-connection v0.1.0
   Compiling v_escape v0.7.4
   Compiling slog-stdlog v4.0.0
   Compiling chain-crypto v0.1.0 (/Users/gb/cardano/jormungandr/chain-deps/chain-crypto)
error[E0061]: this function takes 4 parameters but 3 parameters were supplied
   --> /Users/gb/.cargo/registry/src/github.com-1ecc6299db9ec823/slog-stdlog-4.0.0/lib.rs:244:9
    |
244 |         log::__private_api_log(format_args!("{}", lazy), level, &(target, info.module(), info.file(), info.line()));
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected 4 parameters

error: aborting due to previous error

For more information about this error, try `rustc --explain E0061`.
error: could not compile `slog-stdlog`.
warning: build failed, waiting for other jobs to finish...
error: failed to compile `jormungandr v0.8.2 (/Users/gb/cardano/jormungandr/jormungandr)`, intermediate artifacts can be found at `/Users/gb/cardano/jormungandr/target`

**Comment by eugene-babichenko** [comment]
Have you tried `git submodule update`?

**Comment by hodlonaut** [comment]
Rinor's suggestion to use

cargo build --offline --frozen --bin jormungandr --bin jcli

seems to have gotten me over the line for the time being.

**Comment by rinor** [comment]
Probably due to this https://crates.io/crates/log.
- 0.4.8 (Jul 28, 2019) -  works fine
- 0.4.9 (Dec 15, 2019) - **yanked**
- 0.4.10 (Dec 16, 2019) - released today (approx. 8 hours ago)

**Comment by mzabaluev** [comment]
Try `cargo update` and if the build succeeds, submit the updated `Cargo.lock` in a PR.

**Comment by FonzySF** [comment]
I have the same problem compling on CentsOS7. Tried all the above fixes, but nothing worked.

**Comment by rinor** [comment]
> Try `cargo update` and if the build succeeds, submit the updated `Cargo.lock` in a PR.

I doesn't. It looks like slog-stdlog-4.0.0 broke.

```sh
   Compiling slog-stdlog v4.0.0
error[E0061]: this function takes 4 parameters but 3 parameters were supplied
   --> /home/rinor/.cargo/registry/src/github.com-1ecc6299db9ec823/slog-stdlog-4.0.0/lib.rs:244:9
    |
244 |         log::__private_api_log(format_args!("{}", lazy), level, &(target, info.module(), info.file(), info.line()));
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected 4 parameters

error: aborting due to previous error

For more information about this error, try `rustc --explain E0061`.
error: could not compile `slog-stdlog`.
warning: build failed, waiting for other jobs to finish...
error: build failed
```
the `__private_api_log` signature was changed in https://github.com/rust-lang/log/commit/17063f07d6917383db2579a6d3af1b6f76ed17c7#diff-b4aea3e418ccdb71239b96952d9cddb6

from log 0.4.8:
```
pub fn __private_api_log(
    args: fmt::Arguments,
    level: Level,
    &(target, module_path, file, line): &(&str, &'static str, &'static str, u32),
) 
```

to 0.4.10
```
pub fn __private_api_log(
    args: fmt::Arguments,
    level: Level,
    &(target, module_path, file, line): &(&str, &'static str, &'static str, u32),
    kvs: Option<&[(&str, &str)]>,
)
```

**Comment by rinor** [comment]
Once https://github.com/slog-rs/stdlog/pull/13 is merged, also this issue will be solved.

**Comment by thestarfire** [comment]
Anyone have any ideas?  Also tried all of the suggestions and compile fails on Mohave.   Total newb BTW first post in GIT lol so is this the right place to post or should I start a new issue?

Installing jormungandr v0.8.2 (/Desktop/Jor/jormungandr)
    Updating crates.io index
error: failed to compile `jormungandr v0.8.2 (/Desktop/Jor/jormungandr)`, intermediate artifacts can be found at /Desktop/Jor/target`

Caused by:
  failed to load source for a dependency on `cardano-legacy-address`

Caused by:
  Unable to update /Desktop/Jor/chain-deps/cardano-legacy-address

Caused by:
  failed to read /Desktop/Jor/chain-deps/cardano-legacy-address/Cargo.toml`

**Comment by rinor** [comment]
Is this still an issue or can be closed?

**Comment by hodlonaut** [comment]
Can be closed - closing.
